configuration/parser.go

```diff
@@ -205,6 +206,25 @@ func (p *Parser) overwriteStruct(v reflect.Value, fullpath string, path []string
 	byUpperCase := make(map[string]int)
 	for i := 0; i < v.NumField(); i++ {
 		sf := v.Type().Field(i)
+
+		// For fields inlined in the YAML configuration file, the environment variables also need to be inlined
+		// Example struct tag for inlined field: `yaml:",inline"`
+		yamlTag := sf.Tag.Get("yaml")
+		split := strings.Split(yamlTag, ",")
+		// Skip the first entry, which is the field name
+		isInlined := slices.Contains(split[1:], "inline")
+
+		if isInlined && sf.Type.Kind() == reflect.Struct {
+			// Inlined struct, check whether the env variable corresponds to a field inside this struct
+			// Maps could also be inlined, but since we don't need it right now it is not supported
+			inlined := v.Field(i)
+			for j := 0; j < inlined.NumField(); j++ {
+				if strings.ToUpper(inlined.Type().Field(j).Name) == path[0] {
```

Comment:
Better to use `Equalfold` for these, as it doesn't allocate;


```suggestion
				if strings.EqualFold(inlined.Type().Field(j).Name, path[0]) {
```